src/components/TodoFilter/TodoFilter.tsx

```diff
@@ -1,17 +1,44 @@
 import React from 'react';
+import { Status, StatusType } from '../../types/Status';
+import { useDispatch } from 'react-redux';
+import { setQuery, setStatus } from '../../features/filter';
+import { useAppSelector } from '../../app/hooks';
+
+export const TodoFilter = () => {
+  const dispatch = useDispatch();
+  const { status, query } = useAppSelector(state => state.filter);
+
+  const showOptions = Object.values(StatusType).map(stat => (
+    <option key={stat} value={stat}>
+      {stat[0].toUpperCase() + stat.slice(1)}
+    </option>
+  ));
+
+  const handleChangeStatus = (event: React.ChangeEvent<HTMLSelectElement>) => {
+    dispatch(setStatus(event.target.value as Status));
```

Comment:
You are casting the event target value to `Status` type. It would be safer to check if the value is actually a valid `Status` before dispatching the action. This will help prevent any unexpected values from being dispatched to the state.